dbms/src/DataStreams/NonJoinedBlockInputStream.cpp

```diff
@@ -115,6 +128,10 @@ NonJoinedBlockInputStream::NonJoinedBlockInputStream(const Join & parent_, const
 
 Block NonJoinedBlockInputStream::readImpl()
 {
+    /// build concurrency is less than non join concurrency,
+    /// just return empty block for extra non joined block input stream read
+    if (index >= parent.getBuildConcurrency())
+        return Block();
```

Comment:
It should not happen because here https://github.com/pingcap/tiflash/blob/8e520c2e5347213a67828ee741ad30f58c8d07fc/dbms/src/DataStreams/NonJoinedBlockInputStream.cpp#L71-L72